src/skeleton/Blocks.scala

```diff
@@ -18,8 +18,8 @@ case class BlockAttachParams(
   parentNode: LogicalTreeNode,
   )(implicit val p: Parameters)
 
-case class BlockDescriptor(
+case class BlockDescriptor[T <: LazyModule](
   name:  String,
-  place: BlockAttachParams => Any)
+  place: BlockAttachParams => T)
 
-case object BlockDescriptorKey extends Field[Seq[BlockDescriptor]](Nil)
+case object BlockDescriptorKey extends Field[Seq[BlockDescriptor[LazyModule]]](Nil)
```

Comment:
I think this could also be a `[T <: LazyModule]` thing, which allows you to bind it to a more specific type later if desired.

Reply:
you can't have type parameters on `object`s